java/src/main/java/ai/rapids/cudf/HostColumnVector.java

```diff
@@ -738,15 +738,49 @@ public ColumnBuilder appendLists(List... inputLists) {
     public ColumnBuilder appendStructValues(List<StructData> inputList) {
       for (StructData structInput : inputList) {
         // one row
-        rows++;
         append(structInput);
       }
       return this;
     }
 
+    private void resizeDataBuffer(int length) {
+      // just for strings we want to throw a real exception if we would overrun the buffer
+      if (data == null) {
+        // since rows can be updated as we go, make data buffer as big as the estimate or the requested length
+        if (!type.isNestedType() && type != DType.STRING) {
+          data = HostMemoryBuffer.allocate(Math.min(Integer.MAX_VALUE, estimatedRows * type.getSizeInBytes()));
+        } else {
+          data = HostMemoryBuffer.allocate(length);
+        }
+        return;
+      }
+      long oldLen = data.getLength();
+      long newLen = Math.max(1, oldLen);
+      while (currentByteIndex + length > newLen) {
+        newLen *= 2;
+      }
+      if (newLen > Integer.MAX_VALUE) {
+        throw new IllegalStateException("A data buffer is not supported over 2GB in size");
+      }
+      if (newLen != oldLen) {
+        // need to grow the size of the buffer.
+        HostMemoryBuffer newData = HostMemoryBuffer.allocate(newLen);
+        try {
+          newData.copyFromHostBuffer(0, data, 0, data.length);
+          data.close();
+          data = newData;
+          newData = null;
+        } finally {
+          if (newData != null) {
+            newData.close();
+          }
+        }
+      }
+    }
+
     private void initAndResizeOffsetBuffer(int currentIndex) {
       if (this.offsets == null) {
-        offsets = HostMemoryBuffer.allocate((rows + 1) * OFFSET_SIZE);
+        offsets = HostMemoryBuffer.allocate((estimatedRows + 1) * OFFSET_SIZE);
```

Comment:
Shouldn't this take into account the currentIndex? or will it always be 0?

Reply:
I think it will be always 0 for this condition to be true.